pkg/deployment/deployment.go

```diff
@@ -31,6 +32,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type deployer interface {
+	Configuration
+	Validation
+	Installation
+	Runner
+}
+
+type Configuration interface {
+	Configure(context.Context, *helper.Helper, client.Object, string, string, dataplanev1beta1.AnsibleEESpec) error
+}
+
+type Validation interface {
+	Validate(context.Context, *helper.Helper, client.Object, string, string, dataplanev1beta1.AnsibleEESpec) error
+}
+
+type Installation interface {
+	Install(context.Context, *helper.Helper, client.Object, string, string, dataplanev1beta1.AnsibleEESpec) error
+}
+
+type Runner interface {
+	Run(context.Context, *helper.Helper, client.Object, string, string, dataplanev1beta1.AnsibleEESpec) error
+}
+
+type networker struct{}
+
+type OS struct{}
+
+type cephClient struct{}
+
```

Comment:
So, moving our Slack conversation onto the PR for consistency.

We could leverage these structs to further simplify this but adding name and label fields. Like:

```
type networker struct {
    Name string
    Label string
```

Then move the deployName and deployLabel into the struct to be reference by `ConditionalDeploy()`

Reply:
Yes and then we can remove `deployLabel` argument from ConditionalDeploy, and also `deployName` should be just a simple action name (eg. `Configure`) without target (eg. `Network`, `OS`).